libhttpserver/content_type_utils.go

```diff
@@ -28,20 +28,22 @@ func (w *contentTypeOverridingResponseWriter) overrideMimeType(
 	// by the frontend WebView.
 	ty := strings.ToLower(mimeType)
 	switch {
-	// First reject anything containing javascript/xml/html.
+	// First anything textual as text/plain.
+	// Includes javascript, html, xml. (note that the type may be e.g. application/xhtml+xml)
 	case strings.Contains(ty, "javascript") ||
 		strings.Contains(ty, "xml") ||
-		strings.Contains(ty, "html"):
+		strings.Contains(ty, "html") ||
+		strings.HasPrefix(ty, "text/"):
 		return "text/plain"
 	// Pass multimedia types through, and pdf too.
 	case strings.HasPrefix(ty, "audio/") ||
 		strings.HasPrefix(ty, "image/") ||
 		strings.HasPrefix(ty, "video/") ||
 		ty == "application/pdf":
 		return ty
-	// Otherwise text/plain.
+	// Otherwise default to binary data.
 	default:
-		return "text/plain"
+		return "application/octet-stream"
```

Comment:
Could you explain why we default to binary now? It seems it's safer to default to text to avoid stuff getting executed or parsed by browsers accidentally. If we need to add support for particular types, we can add them explicitly right?